platform/fabricx/core/finality/listenermanager.go

```diff
@@ -64,9 +88,13 @@ func newNotifi(ctx context.Context, cfg config.ConfigService) (*notificationList
 	if err != nil {
 		return nil, err
 	}
+	// CRITICAL FIX: Use context.Background() for the stream lifecycle.
+	// This ensures the stream stays alive for the duration of the shared manager,
+	// regardless of when the individual view's context (ctx) is cancelled.
+	streamCtx := context.Background()
```

Comment:
Good catch. We don't want our notification listener attached to the view's lifecycle. Instead, we want some context that is associated with the lifecycle of the fsc node. That is, when we shutdown the fsc node, the notification listener is stopped gracefully.

Reply:
Maybe we provide implement a start/stop interface for `notificationListenerManager` instead of `Listen()`?